lib/spack/spack/solver/concretize.lp

```diff
@@ -472,6 +472,31 @@ error(2, "Attempted to use external for '{0}' which does not satisfy any configu
 #defined external_spec_condition/4.
 #defined external_spec_condition/5.
 
+%-----------------------------------------------------------------------------
+% Config required semantics
+%-----------------------------------------------------------------------------
```

Comment:
Both `condition_group` and `condition_group_member` need to be `#defined` to avoid warnings like:
```
/home/culpo/PycharmProjects/spack/lib/spack/spack/solver/concretize.lp:505:24-52: info: atom does not occur in any rule head:
  condition_group_member(Y,X)

/home/culpo/PycharmProjects/spack/lib/spack/spack/solver/concretize.lp:505:73-100: info: atom does not occur in any rule head:
  condition_group(Package,X)

/home/culpo/PycharmProjects/spack/lib/spack/spack/solver/concretize.lp:509:3-30: info: atom does not occur in any rule head:
  condition_group(Package,X)

/home/culpo/PycharmProjects/spack/lib/spack/spack/solver/concretize.lp:510:3-31: info: atom does not occur in any rule head:
  condition_group_member(Y,X)

/home/culpo/PycharmProjects/spack/lib/spack/spack/solver/concretize.lp:515:3-30: info: atom does not occur in any rule head:
  condition_group(Package,X)

/home/culpo/PycharmProjects/spack/lib/spack/spack/solver/concretize.lp:521:53-81: info: atom does not occur in any rule head:
  condition_group_member(Y,X)

/home/culpo/PycharmProjects/spack/lib/spack/spack/solver/concretize.lp:521:98-125: info: atom does not occur in any rule head:
  condition_group(Package,X)
```

Reply:
This is fixed by https://github.com/spack/spack/pull/27987/commits/4ae6dcd5221fea7a0081e5e1a9b614f9479dead6